src/utils/cosmosdb_model_versioned.ts

```diff
@@ -75,13 +76,16 @@ export const incVersion = (version: NonNegativeInteger) =>
 export abstract class CosmosdbModelVersioned<
   T,
   TN extends Readonly<T & Partial<NewVersionedModel>>,
-  TR extends Readonly<T & RetrievedVersionedModel>
-> extends CosmosdbModel<T, TN & BaseModel, TR> {
+  TR extends Readonly<T & RetrievedVersionedModel>,
+  IDENTITY extends keyof T,
```

Comment:
if this is the modelId can e name it 'ModelId' ?

Reply:
well, Modeld stands for Model Identity, and we are in the context of a model, so....

I can rename it, but I'll also have to deal with `ModelId` which is a decoder that maps an id field value (just a string decoder so far).